[Ho Jian Tao] iP

jiantaooo opened the pull request:
### 👑✨ **Let the Princess Manage Your Tasks!** ✨👑

🌸Welcome to a magical world where the Princess will take care of your tasks! This Task Management Application will help you manage your tasks easily and gracefully. 🌸


### **💖 How It Works**

- The Princess types commands like todo, deadline, or event to add tasks.
- Tasks are shown and organized, with options to mark them as done or delete them.
- Before the Princess finishes her day (when the app closes), she saves all the tasks so they’re ready for tomorrow.


### **🌟 Features for You**
- [ ]  Add tasks
- [ ]  Mark tasks as complete or undone
- [ ]  Delete tasks that are no longer needed
- [ ]  Search for tasks with a keyword
- [ ]  Tasks are safely saved and loaded from a file
- [ ]  Don't know how to command your princess? type 'help' for help!


### **✨Type of tasks**
1. 📝 To-Do Tasks
2. ⏳ Deadline Tasks
3. 📅 Event Tasks



### **👑 Inline Code Example**
To add a new task, type todo <task_name>. It will be added to your list right away!


👉 What are you waiting for? Experience the magic today! Click [here ](https://github.com/Alolononon/ip/archive/refs/heads/master.zip)to Download now and let the Princess handle your tasks!]👑✨






Here’s a small part of the code that shows the main method:
```java
    public static void main(String[] args) {
        String filePath = "data/saved_tasks.txt";
        new Princess(filePath).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jiantaooo committed:
Rename, Greet, Exit

jiantaooo committed:
Echo

jiantaooo committed:
Add, List

jiantaooo committed:
Mark as Done

jiantaooo committed:
ToDos, Events, Deadlines

jiantaooo committed:
Automated Text UI Testing

jiantaooo committed:
Handle Errors

jiantaooo committed:
Delete

jiantaooo committed:
add saved to file function, add read to file function, edit input for testing

jiantaooo committed:
Merge branch 'branch-Level-7'

jiantaooo committed:
add Dates and Times feature for "deadline" and "event" tasks

jiantaooo committed:
Merge branch 'branch-Level-8'

TiewJiaLiang commented on src/main/java/DateParser.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.List;
+
+public class DateParser {
+    private static final List<DateTimeFormatter> DATE_FORMATS = List.of(
```
Great job in following the standard for constant!

TiewJiaLiang commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+
+public class Event extends Task {
+
+    protected LocalDateTime from;
```
Good use of simple naming conventions!

TiewJiaLiang commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+        this.to = DateParser.parseDateTime(to);
+    }
+
+    public String getFrom() {
```
Good job in using verbs for methods!

TiewJiaLiang commented on src/main/java/FileStoringRetrieving.java:
```diff
@@ -0,0 +1,110 @@
+                String description = lineArr[2];
+                Task task = null;
+
+                switch (type) {
```
Great job in following the standard for switch statements!

TiewJiaLiang commented on src/main/java/Princess.java:
```diff
@@ -0,0 +1,194 @@
+        while (true) {
+            try {
+                String input = sc.nextLine(); String[] inputss = input.split(" "); String code = inputss[0];
+                Task task;
```
Great job in using nouns for variables!

TiewJiaLiang reviewed (commented):
Amazing job in following the coding standards and naming conventions!

jiantaooo committed:
Add JavaDoc comments

jiantaooo committed:
Revert "Add JavaDoc comments"

This reverts commit 86a6a470bd2b81a06f7227703082472a043c900f.

jiantaooo committed:
Reapply "Add JavaDoc comments"

This reverts commit a13c2c8e9de70850da83634eb6f2c7c8a524e7b5.

jiantaooo committed:
add find function to Give users a way to find a task by searching for a keyword in the task description

jiantaooo committed:
Merge branch 'branch-Level-9'

jiantaooo committed:
add princess.java documentation for javadoc

jiantaooo committed:
add 'find' command into help function

jiantaooo committed:
Merge branch 'A-JavaDoc'

jiantaooo committed:
Merge branch 'branch-Level-9'

luna-ortus-cor commented on src/main/java/DateParser.java:
```diff
@@ -0,0 +1,60 @@
+            DateTimeFormatter.ofPattern("dd/MM/yyyy h:mma"),    // 01/03/2000 4pm
+            DateTimeFormatter.ofPattern("MMM dd yyyy, h:mma")  // Mar 01 2000, 4pm
+
+    );
```
Good use of constant to handle extensive range of datetime formats.

luna-ortus-cor commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime  by;
```
```suggestion
    protected LocalDateTime by;
```

jiantaooo replied:
Thanks for the sharp eye! Appreciate it!!

luna-ortus-cor commented on src/main/java/FileStoringRetrieving.java:
```diff
@@ -0,0 +1,130 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
```
It might be better to not use wildcard imports for better code clarity and to adhere to best coding standards.

luna-ortus-cor commented on src/main/java/FileStoringRetrieving.java:
```diff
@@ -0,0 +1,130 @@
+                        task = null; // Ensure valid event format
+                    task = new Event(description, lineArr[3], lineArr[4]); // parts[3] = from, parts[4] = to
+                    break;
+                }
```
It would be good to include a ```default``` case to handle any unexpected cases e.g. in the event of a bad write or corrupted file.

jiantaooo replied:
Great point! I'll add a default case to handle unexpected situations!

luna-ortus-cor commented on src/main/java/Princess.java:
```diff
@@ -0,0 +1,220 @@
+import java.io.IOException;
+import java.util.*;
```
It would be good to not use wildcard imports.

jiantaooo replied:
oops... thanks!

luna-ortus-cor commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+  }
```
```suggestion
    }
```

luna-ortus-cor reviewed (commented):
Overall looks quite good, some minor coding style violations. Consider moving the ```System.out.println``` calls to a separate class to handle that and minimize clutter in the main files (especially ```Princess.java```).

jiantaooo committed:
edit into more OOP format

jiantaooo committed:
Organize the classes into suitable java packages.

jiantaooo committed:
change runtest.bat

jiantaooo reviewed (commented):
thanks for your suggestions. great help!

jiantaooo committed:
add gradle support

jiantaooo committed:
Merge branch 'add-gradle-support'

jiantaooo committed:
Add JUnit tests (ToDoTest.java and UITest.java)

jiantaooo committed:
Merge branch 'A-JUnit'

jiantaooo committed:
prepare to package the App as a JAR file

jiantaooo committed:
edit Command.java to be more readable and structured

jiantaooo committed:
Merge branch 'add-gradle-support' of https://github.com/Alolononon/ip

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

jiantaooo committed:
Merge branch 'add-gradle-support' of https://github.com/Alolononon/ip into add-gradle-support

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

jiantaooo committed:
Merge branch 'add-gradle-support'

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.properties

jiantaooo committed:
add checkstyle

jiantaooo committed:
Merge branch 'A-CheckStyle'

jiantaooo committed:
add javadoc for Princess.java

jiantaooo committed:
Add a GUI to the chatbot

jiantaooo committed:
Merge branch 'GUI-branch'

jiantaooo committed:
update on unit testing based on UI change

jiantaooo committed:
Task class: Add assertions to ensure task name validity

The Task class does not validate the task name before assigning it,
which could lead to unexpected null or empty task names being stored.

If a task is created with a null or empty name, it may cause issues
when retrieving or displaying the task.

Let's add assertions to:
* Ensure that the task name is not null or empty when a Task object
  is instantiated.
* Ensure that the task name remains non-null when accessed using
  getTaskName().

Adding these assertions prevents invalid task names from being stored
or retrieved, improving overall system robustness.

jiantaooo committed:
Event class: Add assertions to validate event task data

The Event class does not validate its inputs before parsing the
start and end date/time, which could lead to errors if null or
empty values are provided.

If an Event object is created with invalid inputs, it may cause
unexpected behavior or exceptions when retrieving or formatting
date/time values.

Let's add assertions to:
* Ensure that the event description is not null or empty.
* Ensure that the event start and end times are not null or empty
  before parsing them.
* Ensure that the parsed LocalDateTime objects are not null when
  retrieving formatted values.

Adding these assertions helps prevent invalid event creation and
improves system robustness by catching incorrect inputs early.

jiantaooo committed:
add assertion into multiple class

jiantaooo committed:
add CommandType
refactor Command

jiantaooo committed:
Merge branch 'branch-A-Assertions'

jiantaooo committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/princess/command/Command.java

jiantaooo committed:
Merge branch 'branch-A-CodeQuality'

jiantaooo committed:
implement add place function when creating task

jiantaooo committed:
add javadoc on multiple files
fix checkstyle problem

jiantaooo committed:
bug fix: Accept both capitalized and uncapitalized date/time inputs

jiantaooo committed:
Add 2-second delay before closing window on "bye" command

jiantaooo committed:
add picture for user guide
add references in README.md

jiantaooo committed:
update README.md in docs folder

jiantaooo committed:
change Ui photo for website

jiantaooo committed:
fix bug:
- catch error when loading file
- error loading date format from file

jiantaooo committed:
edit on comments
